src/UBAFeeCalculator/UBAFeeUtility.ts

```diff
@@ -19,14 +19,17 @@ export function performLinearIntegration(
   integralStart: BigNumber,
   integralEnd: BigNumber
 ): BigNumber {
+  const scaler = parseEther("1.0");
   const lengthUnderCurve = integralEnd.sub(integralStart);
   const resolveValue = (index: number): BigNumber => cutoffArray[index][1];
-  let feeIntegral = resolveValue(Math.min(index, cutoffArray.length - 1)).mul(lengthUnderCurve);
+  let feeIntegral = resolveValue(index === 0 ? 0 : index === cutoffArray.length ? cutoffArray.length - 1 : index - 1)
```

Comment:
This is a bit hard to read, can you add a comment or refactor with more intermediate variables to understand how to change this in the future?

Reply:
Good call - I've done both (simplified & added documentation). In the end, this could be reduced to a simple

`MAX(0, MIN(INDEX - 1, ARR_LENGTH))`

Updated: https://github.com/across-protocol/sdk-v2/pull/257/commits/f606f38fdfdff158c6a1aeb98f1ed14d11b725b8